app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -52,4 +52,19 @@ ManageIQ.angular.app.controller('cloudSubnetFormController', ['$http', '$scope',
     $scope.angularForm.$setPristine(true);
     miqService.miqFlash("warn", "All changes have been reset");
   };
+
+  $scope.filterNetworkManagerChanged = function(id) {
+    if (id) {
+      miqService.sparkleOn();
+      $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id).success(function(data) {
+        $scope.available_networks = data.available_networks;
+      }).then(function(id) {
+        $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id).success(function(data) {
```

Comment:
Also also, since those 2 are not even dependent, they should be a parallel request, not serial:

```js
$scope.filterNetworkManagerChanged = function(id) {
  if (! id) {
    return;
  }

  miqService.sparkleOn();
  $q.all([
    $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id),
    $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id),
  ])
    .then(function(data) {
      $scope.available_networks = data[0].available_networks;
      $scope.available_tenants = data[1].available_tenants;
    })
    .then(function() {
      miqService.sparkleOff();
    });
};
```

Reply:
Thanks